### PR TITLE
Update documentation to version 0.4.0.

### PR DESCRIPTION
**Pull request by chriseth**
This applies some changes that happened with version 0.4.0 and also includes other cleanups.

### CODE REVIEW

**Comment by chriseth** [comment]
@Denton-L if you would like to have a look here

File: docs/control-structures.rst [hunk 1]
```diff
@@ -337,8 +353,9 @@ arising when writing manual assembly by the following features:
 We now want to describe the inline assembly language in detail.
 
 .. warning::
-    Inline assembly is still a relatively new feature and might change if it does not prove useful,
-    so please try to keep up to date.
+    Inline assembly is a way to access the Ethereum Virtual Machine
+    at a low level. This discards and allows you to bypass several safety
```
Comment (axic) [inline docs/control-structures.rst]: `discards and allows you bypass` is repetitive. Probably `This discards several important safety` is enough.

File: docs/security-considerations.rst [hunk 2]
```diff
@@ -134,6 +134,11 @@ Sending and Receiving Ether
      means for the recipient to block progress in the sending contract. Again, the best practice here is to use
      a :ref:`"withdraw" pattern instead of a "send" pattern <withdrawal_pattern>`.
 
+- Contracts currently cannot prevent that someone sends them Ether.
```
Comment (axic) [inline docs/security-considerations.rst]: Maybe it makes sense highlighting this somehow?

File: docs/security-considerations.rst [hunk 3]
```diff
@@ -155,6 +160,7 @@ Never use tx.origin for authorization. Let's say you have a wallet contract like
 
     pragma solidity ^0.4.0;
 
+    // THIS CONTRACT CONTAINS A BUG - DO NOT USE
```
Comment (axic) [inline docs/security-considerations.rst:163]: Maybe make it a bit more to stand out?

File: docs/security-considerations.rst [hunk 4]
```diff
@@ -186,7 +192,7 @@ Now someone tricks you into sending ether to the address of this attack wallet:
         }
     }
 
-If your wallet had checked msg.sender for authorization, it would get the address of the attack wallet, instead of the owner address. But by checking tx.origin, it gets the original address that kicked off the transaction, which is still the owner address. The attack wallet instantly drains all your funds.
+If your wallet had checked ``msg.sender`` for authorization, it would get the address of the attack wallet, instead of the owner address. But by checking tx.origin, it gets the original address that kicked off the transaction, which is still the owner address. The attack wallet instantly drains all your funds.
```
Comment (axic) [inline docs/security-considerations.rst]:
```
``tx.origin``
```

Comment (axic) [inline docs/security-considerations.rst]: `attack wallet` -> `attacker wallet` or `attacker's wallet`?

File: docs/security-considerations.rst (hunk 2 above)
Comment (chriseth) [reply docs/security-considerations.rst]: Moved it to the top

File: docs/security-considerations.rst (hunk 3 above)
Comment (chriseth) [reply docs/security-considerations.rst:163]: I think this is pretty visible.

File: docs/security-considerations.rst (hunk 4 above)
Comment (chriseth) [reply docs/security-considerations.rst]: I think it's fine, we don't have to draw a person into this :-)

**Pull request merged**